extensions-contrib/aliyun-oss-extensions/src/main/java/org/apache/druid/data/input/aliyun/OssInputSource.java

```diff
@@ -0,0 +1,178 @@
+   * @param uris              User provided uris to read input data
+   * @param prefixes          User provided prefixes to read input data
+   * @param objects           User provided cloud objects values to read input data
+   * @param inputSourceConfig User provided properties for overriding the default aliyun-oss configuration
+   */
+  @JsonCreator
+  public OssInputSource(
+      @JacksonInject OSS client,
+      @JacksonInject OssInputDataConfig inputDataConfig,
+      @JsonProperty("uris") @Nullable List<URI> uris,
+      @JsonProperty("prefixes") @Nullable List<URI> prefixes,
+      @JsonProperty("objects") @Nullable List<CloudObjectLocation> objects,
+      @JsonProperty("properties") @Nullable OssClientConfig inputSourceConfig
+  )
+  {
+    super(OssStorageDruidModule.SCHEME, uris, prefixes, objects);
+    this.inputDataConfig = Preconditions.checkNotNull(inputDataConfig, "inputDataConfig");
+    Preconditions.checkNotNull(client, "client");
+    this.inputSourceConfig = inputSourceConfig;
+    this.clientSupplier = Suppliers.memoize(
+        () -> {
+          if (inputSourceConfig != null) {
+            return inputSourceConfig.buildClient();
+          } else {
+            return client;
+          }
+        }
+    );
+  }
+
+
+  @Nullable
+  @JsonProperty("properties")
+  public OssClientConfig getOssInputSourceConfig()
+  {
+    return inputSourceConfig;
+  }
+
+  @Override
+  protected InputEntity createEntity(CloudObjectLocation location)
+  {
+    return new OssEntity(clientSupplier.get(), location);
+  }
+
+  @Override
+  protected Stream<InputSplit<List<CloudObjectLocation>>> getPrefixesSplitStream(@Nonnull SplitHintSpec splitHintSpec)
+  {
+    final Iterator<List<OSSObjectSummary>> splitIterator = splitHintSpec.split(
+        getIterableObjectsFromPrefixes().iterator(),
+        object -> new InputFileAttribute(object.getSize())
+    );
+
+    return Streams.sequentialStreamFrom(splitIterator)
+                  .map(objects -> objects.stream()
+                                         .map(OssUtils::summaryToCloudObjectLocation)
+                                         .collect(Collectors.toList()))
+                  .map(InputSplit::new);
+  }
+
+  @Override
+  public SplittableInputSource<List<CloudObjectLocation>> withSplit(InputSplit<List<CloudObjectLocation>> split)
+  {
+    return new OssInputSource(
+        clientSupplier.get(),
+        inputDataConfig,
+        null,
+        null,
+        split.get(),
+        getOssInputSourceConfig()
+    );
+  }
+
+  @Override
+  public int hashCode()
+  {
+    return Objects.hash(super.hashCode(), inputSourceConfig);
+  }
+
+  @Override
+  public boolean equals(Object o)
+  {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    if (!super.equals(o)) {
+      return false;
+    }
+    OssInputSource that = (OssInputSource) o;
```

Comment:
I think equals/hashcode should have uris, prefixes, etc.

Reply:
it's `super.equals` which checks uris and prefixes